librz/core/cmd.c

```diff
@@ -6007,17 +6054,22 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(html_enable_stmt) {
 }
 
 DEFINE_HANDLE_TS_FCN_AND_SYMBOL(pipe_stmt) {
-	TSNode first_cmd = ts_node_named_child(node, 0);
-	rz_return_val_if_fail(!ts_node_is_null(first_cmd), false);
-	TSNode second_cmd = ts_node_named_child(node, 1);
-	rz_return_val_if_fail(!ts_node_is_null(second_cmd), false);
-	char *first_str = ts_node_sub_string(first_cmd, state->input);
-	char *second_str = ts_node_sub_string(second_cmd, state->input);
-	int value = state->core->num->value;
-	RzCmdStatus res = rz_cmd_int2status(rz_core_cmd_pipe(state->core, first_str, second_str));
-	state->core->num->value = value;
-	free(first_str);
-	free(second_str);
+	TSNode command_rizin = ts_node_named_child(node, 0);
+	TSNode command_pipe = ts_node_named_child(node, 1);
+
+	char *rz_cmd = ts_node_sub_string(command_rizin, state->input);
+	if (!rz_cmd) {
+		return RZ_CMD_STATUS_INVALID;
+	}
+
+	RzCmdStatus res = RZ_CMD_STATUS_INVALID;
+	RzCmdParsedArgs *a = ts_node_handle_arg_prargs(state, node, command_pipe, 1, true);
+	if (a && a->argc > 1) {
```

Comment:
I would check `if (!a || a->argc <= 1) return INVALID`. Otherwise you just ignore the error, which I think is never good.

Reply:
that is not true. `res` is `RZ_CMD_STATUS_INVALID` so if that `if` fails, it will return `RZ_CMD_STATUS_INVALID`

Reply:
Right!